packages/flutter_tools/lib/src/base/dds.dart

```diff
@@ -52,6 +53,8 @@ class DartDevelopmentService {
           serviceUri: ddsUri,
           enableAuthCodes: disableServiceAuthCodes != true,
           ipv6: ipv6 == true,
+          // Enables caching of CPU samples collected during application startup.
+          cachedUserTags: const <String>['AppStartUp'],
```

Comment:
Is there any documentation of supported tags that we could link to?

Reply:
In theory, this works for any set `UserTag`, so this is something that could be exposed as a CLI option at some point so users can cache samples associated with any `UserTag` defined in their code. However, I think it's best to just start with this single case so we can make progress on tooling for profiling application startup times.

Reply:
Oh, I see, so `UserTag`s are defined in user code?

Reply:
Yup, but we're setting this particular tag in the engine and resetting to default in the framework after everything's initialized. See [UserTag](https://api.dart.dev/stable/2.13.4/dart-developer/UserTag-class.html).